projects/client/RabbitMQ.Client/src/client/api/AmqpTcpEndpoint.cs

```diff
@@ -180,9 +182,18 @@ public IProtocol Protocol
         public System.Net.Sockets.AddressFamily AddressFamily { get; set; } = ConnectionFactory.DefaultAddressFamily;
 
         /// <summary>
-        /// Retrieve the SSL options for this AmqpTcpEndpoint. If not set, null is returned.
+        /// Retrieve the SSL options for this AmqpTcpEndpoint. If not set, a default SslOption instance is returned.
         /// </summary>
-        public SslOption Ssl { get; set; }
+        public SslOption Ssl
+        {
+            get { return _ssl; }
+            set
+            {
+                if (_ssl == null)
```

Comment:
This does not match the summary description. Shouldn't this be checking the incoming value to ensure that the property is not set to null? If so, the exception should be properly changed to reflect that as well.
```cs
if (value == null)
    throw new ArgumentNullException(nameof(value), "Ssl property must be an instance of SslOption (cannot be null)");
```